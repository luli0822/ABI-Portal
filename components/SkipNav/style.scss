@import "../../scss/a11y-mixins";

$rootClass: "skip-nav";

.#{$rootClass} {
  display: flex;
  justify-content: center;

  &__link {
    overflow: hidden;
    height: 1px;
    margin: -1px;

    &:focus {
      @include focus-inset($white);

      // position absolute is needed here to override the position: relative in the focus mixin
      position: absolute;
      background-color: $primary-color;
      padding: 10px;
      color: $white;
      z-index: 9999;
      text-align: center;
      height: auto;
      min-width: 220px;
      top: 10px;
    }
  }
}
