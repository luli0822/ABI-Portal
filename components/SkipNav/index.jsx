import React from 'react';
import PropTypes from 'prop-types';
import './style.scss';

function SkipNav(props) {
	const { skipToLinkText } = props;
	const rootClass = 'skip-nav';
	return (
		<div className={rootClass}>
			<a href="#main" className={`${rootClass}__link`}>
				{skipToLinkText}
			</a>
		</div>
	);
}

SkipNav.propTypes = {
	skipToLinkText: PropTypes.string.isRequired,
};

export default SkipNav;
