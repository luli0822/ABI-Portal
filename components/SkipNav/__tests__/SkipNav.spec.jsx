import React from 'react';
import { shallow } from 'enzyme';
import SkipNav from '../index';

describe('<SkipNav />', () => {
	it('renders correctly', () => {
		expect(shallow(<SkipNav skipToLinkText="Skip to main content" />).debug()).toMatchSnapshot();
	});
});
