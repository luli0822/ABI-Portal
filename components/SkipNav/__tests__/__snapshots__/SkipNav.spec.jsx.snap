// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkipNav /> renders correctly 1`] = `
"<div className=\\"skip-nav\\">
  <a href=\\"#main\\" className=\\"skip-nav__link\\">
    Skip to main content
  </a>
</div>"
`;
